### ISSUE TITLE
[CKV_AWS_358] - false alarm triggered even when GH Organisation is specified

### ISSUE THREAD

**Issue by cbugneac-nex** [bridgecrewio/checkov#5423]
**Describe the issue**

I guess there is false alarm triggered by this check.
The GitHub Organisation is already included in the values of condition for the `token.actions.githubusercontent.com:sub` variable.

**Examples**
```terraform
data "aws_iam_policy_document" "assumerole" {
  statement {
    sid    = "DefaultAssumeRole"
    effect = "Allow"

    actions = [
      "sts:AssumeRoleWithWebIdentity",
    ]

    condition {
      test     = "StringEquals"
      variable = "token.actions.githubusercontent.com:aud"

      values = [
        "sts.amazonaws.com"
      ]
    }

    condition {
      test     = "StringLike"
      variable = "token.actions.githubusercontent.com:sub"

      values = [
        "repo:GitHub-Organization/${var.github_repo}:*"
      ]
    }

    principals {
      type = "Federated"
      identifiers = [
        "arn:aws:iam::${var.aws_account_id}:oidc-provider/token.actions.githubusercontent.com"
      ]
    }
  }
}
```

Checkov output:
```sh
Check: CKV_AWS_358: "Ensure GitHub Actions OIDC trust policies only allows actions from a specific known organization"
	FAILED for resource: module.ecr_repo.aws_iam_policy_document.assumerole
	File: /../../modules/ecr-repository/data_aws_iam_policy_document_assumerole.tf:1-35
	Calling File: /module_ecr_repository.tf:2-24
```

**Version:**
 - Checkov Version [2.3.359]

**Additional context**
See https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services

**Comment by gruebel** [comment]
hey @cbugneac-nex thanks for reaching out.

I found the issue. It is strange that exactly the same resource passes, when it next to other resources, but as a standalone not 😅

**Comment by cbugneac-nex** [comment]
Hi, please can this be re-opened ?
with version 3.0.40 I'm still getting this false alarm.

![image](https://github.com/bridgecrewio/checkov/assets/123649699/dacf3298-d31e-42e0-b501-c123c012f82a)
